js/apps/system/_admin/aardvark/APP/react/src/components/form/FormField.tsx

```diff
@@ -15,6 +17,9 @@ export type FormFieldProps = {
   group?: string;
   isRequired?: boolean;
   isDisabled?: boolean;
+  isClearable?: boolean;
+  placeholder?: string;
+  noOptionsMessage?: ((obj: { inputValue: string }) => ReactNode) | undefined;
 };
 
```

Comment:
You could extract some props and selectProps that the cases in the switch have in common, for example:

```
const commonProps = {
    isDisabled: field.isDisabled,
    isRequired: field.isRequired,
    name: field.name
  };
  
const selectProps = {
    autoFocus,
    options: field.options,
    noOptionsMessage: field.noOptionsMessage,
    placeholder: field.placeholder
  };
```

And then apply it to the correspondent components at the switch cases:

`<SwitchControl {...commonProps} />`
```<InputControl
            {...commonProps}
            inputProps={{
              type: "number",
              autoFocus,
              placeholder: field.placeholder
            }}
          />```
`<MultiSelectControl {...commonProps} selectProps={selectProps} />`
```
<CreatableSingleSelectControl
            isDisabled={field.isDisabled}
            selectProps={{...selectProps, isClearable: field.isClearable }}
            isRequired={field.isRequired}
            name={field.name}
          />
```

Reply:
Good suggestion, have refactored this internally without changing the outside API